src/rust/cryptography-x509-validation/src/types.rs

```diff
@@ -110,9 +112,144 @@ impl<'a> DNSPattern<'a> {
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum IPAddress {
+    V4(std::net::Ipv4Addr),
+    V6(std::net::Ipv6Addr),
```

Comment:
Is there an advantage or disadvantage to being an enum, rather than wrapping `IpAddr` which is already an enum?

Reply:
None that I can think of.

Reply:
I probably suggested it, thinking that `FromStr` wasn't implemented on `IpAddr`. But I was wrong :slightly_smiling_face: